Mathlib/CategoryTheory/Limits/Shapes/WideEqualizers.lean

```diff
@@ -0,0 +1,836 @@
+
+/-- A cotrident `t` on the parallel family `f : J → (X ⟶ Y)` consists of two morphisms
+    `t.ι.app zero : X ⟶ t.X` and `t.ι.app one : Y ⟶ t.X`. Of these, only the second one is
+    interesting, and we give it the shorter name `Cotrident.π t`. -/
+abbrev Cotrident.π (t : Cotrident f) :=
+  t.ι.app one
+#align category_theory.limits.cotrident.π CategoryTheory.Limits.Cotrident.π
+
+@[simp]
+theorem Trident.ι_eq_app_zero (t : Trident f) : t.ι = t.π.app zero :=
+  rfl
+#align category_theory.limits.trident.ι_eq_app_zero CategoryTheory.Limits.Trident.ι_eq_app_zero
+
+@[simp]
+theorem Cotrident.π_eq_app_one (t : Cotrident f) : t.π = t.ι.app one :=
+  rfl
+#align category_theory.limits.cotrident.π_eq_app_one CategoryTheory.Limits.Cotrident.π_eq_app_one
+
+@[reassoc (attr := simp)]
+theorem Trident.app_zero (s : Trident f) (j : J) : s.π.app zero ≫ f j = s.π.app one := by
+  rw [← s.w (line j), parallelFamily_map_left]
+#align category_theory.limits.trident.app_zero CategoryTheory.Limits.Trident.app_zero
+
+@[reassoc (attr := simp)]
+theorem Cotrident.app_one (s : Cotrident f) (j : J) : f j ≫ s.ι.app one = s.ι.app zero := by
+  rw [← s.w (line j), parallelFamily_map_left]
+#align category_theory.limits.cotrident.app_one CategoryTheory.Limits.Cotrident.app_one
+
+/-- A trident on `f : J → (X ⟶ Y)` is determined by the morphism `ι : P ⟶ X` satisfying
+`∀ j₁ j₂, ι ≫ f j₁ = ι ≫ f j₂`.
+-/
+@[simps]
+def Trident.ofι [Nonempty J] {P : C} (ι : P ⟶ X) (w : ∀ j₁ j₂, ι ≫ f j₁ = ι ≫ f j₂) : Trident f
+    where
+  pt := P
+  π :=
+    { app := fun X => WalkingParallelFamily.casesOn X ι (ι ≫ f (Classical.arbitrary J))
+      naturality := fun i j f => by
+        dsimp
+        cases' f with _ k
+        · simp
+        · simp [w (Classical.arbitrary J) k] }
+#align category_theory.limits.trident.of_ι CategoryTheory.Limits.Trident.ofι
+
+/-- A cotrident on `f : J → (X ⟶ Y)` is determined by the morphism `π : Y ⟶ P` satisfying
+`∀ j₁ j₂, f j₁ ≫ π = f j₂ ≫ π`.
+-/
+@[simps]
+def Cotrident.ofπ [Nonempty J] {P : C} (π : Y ⟶ P) (w : ∀ j₁ j₂, f j₁ ≫ π = f j₂ ≫ π) :
+    Cotrident f where
+  pt := P
+  ι :=
+    { app := fun X => WalkingParallelFamily.casesOn X (f (Classical.arbitrary J) ≫ π) π
+      naturality := fun i j f => by
+        dsimp
+        cases' f with _ k
+        · simp
+        · simp [w (Classical.arbitrary J) k] }
+#align category_theory.limits.cotrident.of_π CategoryTheory.Limits.Cotrident.ofπ
+
+-- See note [dsimp, simp]
+theorem Trident.ι_ofι [Nonempty J] {P : C} (ι : P ⟶ X) (w : ∀ j₁ j₂, ι ≫ f j₁ = ι ≫ f j₂) :
+    (Trident.ofι ι w).ι = ι :=
+  rfl
+#align category_theory.limits.trident.ι_of_ι CategoryTheory.Limits.Trident.ι_ofι
+
+theorem Cotrident.π_ofπ [Nonempty J] {P : C} (π : Y ⟶ P) (w : ∀ j₁ j₂, f j₁ ≫ π = f j₂ ≫ π) :
+    (Cotrident.ofπ π w).π = π :=
+  rfl
+#align category_theory.limits.cotrident.π_of_π CategoryTheory.Limits.Cotrident.π_ofπ
+
+-- porting note: @[simp] as the linter complains even if we increase the priority
```

Comment:
This wasn't a `simp` lemma in mathlib3, was it? I'm not sure why we have this porting note.

Reply:
It seems it had tagged an a `simp` lemma earlier in the port process. I have just fixed this.

Reply:
LGTM!